src/coreclr/vm/amd64/asmhelpers.S

```diff
@@ -586,7 +586,21 @@ END_PROLOGUE
         mov             rax, qword ptr [rsp + 8 + 8 + __InterpreterStubArgumentRegistersOffset]
         pop_register    rcx
         ret
-NESTED_END InterpreterStubRetBuff, _TEXT
+NESTED_END InterpreterStubRetBuffRDI, _TEXT
+
+NESTED_ENTRY InterpreterStubRetBuffRSI, _TEXT, NoHandler
+        push_register   rax // stack alignment
+END_PROLOGUE
+        mov             rdi, rax // pTransitionBlock*
+        mov             rsi, rbx // the IR bytecode pointer
+        // Load the return buffer address
+        // The 8 + 8 is for the push and the return address slot
+        mov             rdx, qword ptr [rsp + 8 + 8 + __InterpreterStubArgumentRegistersOffset + 8]
```

Comment:
Is the "+ 8" the only important difference here? I assume we are simply pairing the use of `rdi` and `rsi` for the interpreter with the JIT.